src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,267 @@
+        integrations.executeOnKey(integrationName, UpdateIntegrationEntryProcessor(integrationUpdate))
+    }
+
+    fun deleteIntegrationDefinition(integrationName: String) {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
+        val integration = integrations.getValue(integrationName)
+        integrations.remove(integrationName)
+        entitySetManager.deleteEntitySet(integration.logEntitySetId.get())
+    }
+
+    private fun getSrcDataSource(source: Properties): HikariDataSource {
+        return HikariDataSource(HikariConfig(source))
+    }
+
+    private fun generateDestinationsMap(integration: Integration, missionParameters: MissionParameters, dataIntegrationApi: DataIntegrationApi): Map<StorageDestination, IntegrationDestination> {
+        val s3BucketUrl = integration.s3bucket
+        val dstDataSource = HikariDataSource(HikariConfig(missionParameters.postgres.config))
+        integration.maxConnections.ifPresent { dstDataSource.maximumPoolSize = it }
+
+        val pgDestination = PostgresDestination(
+                entitySets.mapKeys { it.value.id },
+                entityTypes,
+                propertyTypes.mapKeys { it.value.id },
+                dstDataSource
+        )
+
+        if (s3BucketUrl.isNotBlank()) {
+            val s3Api = Retrofit.Builder()
+                    .baseUrl(s3BucketUrl)
+                    .addConverterFactory(RhizomeByteConverterFactory())
+                    .addConverterFactory(RhizomeJacksonConverterFactory(ObjectMappers.getJsonMapper()))
+                    .addCallAdapterFactory(RhizomeCallAdapterFactory())
+                    .client(RetrofitBuilders.okHttpClient().build())
+                    .build().create(S3Api::class.java)
+            val s3Destination = PostgresS3Destination(pgDestination, s3Api, dataIntegrationApi)
+            return mapOf(StorageDestination.POSTGRES to pgDestination, StorageDestination.S3 to s3Destination)
+        }
+        return mapOf(StorageDestination.POSTGRES to pgDestination)
+    }
+
+    private fun createLogEntitySet(integrationName: String, integration: Integration): UUID {
+        val name = buildLogEntitySetName(integrationName)
+        val contacts = integration.contacts
+        val description = buildLogEntitySetDescription(integration.flightPlanParameters)
+        val logEntitySet = EntitySet(
+                logEntityType.id,
+                name,
+                name,
+                Optional.of(description),
+                contacts
+        )
+        return entitySetManager.createEntitySet(Principals.getCurrentUser(), logEntitySet)
+    }
+
+    private fun buildLogEntitySetName(integrationName: String): String {
+        var name = "Integration logs for $integrationName"
+        var nameAttempt = name
+        var count = 1
+
+        while (reservationService.isReserved(nameAttempt)) {
+            nameAttempt = name + "_" + count
```

Comment:
super minor but I prefer the kotlin string building of `nameAttempt = "${name}_$count`